### ISSUE TITLE
AbortError thrown during reconnect attempts

### ISSUE THREAD

**Issue by irakliy81** [NodeRedis/node_redis#1242]
* **Version**: node_redis 2.7.1
* **Platform**: Node.js 6.9.0 / Azure

I have defined a simple connection retry strategy that looks something like this:
```JavaScript
function(options: RetryStrategyOptions) {
    if (options.error && options.error.code === 'ECONNREFUSED') {
        return new Error('The server refused the connection');
    }
    else if (options.total_retry_time > MAX_RETRY_TIME) {
        return new Error('Retry time exhausted');
    }

    return Math.min(options.attempt * RETRY_INTERVAL_STEP, MAX_RETRY_INTERVAL);
}
```

However, the client still throws errors on reconnect attempts (which results in unhandled exception). It seems like the client is trying to execute `AUTH` command on reconnect and is failing. I can get around the error by doing something like this:
```JavaScript
this.client.on('error', (error) => {
    if (error.command === 'AUTH' && error.code === 'UNCERTAIN_STATE') {                
        // this will be triggered on reconnect attempts - do nothing
    }
    else {
        this.emit(ERROR_EVENT, new Error('my custom error'));
    }
});
```
But this seems like a hack to me. Should the client throw errors while it is trying to reconnect?

**Comment by BridgeAR** [comment]
Fixed in 2.8.0

**Comment by irakliy81** [comment]
Great! Thank you!
